docs/key_modules/memory.md

```diff
@@ -0,0 +1,315 @@
+
+Methods:
+
+- `retrieve()`: Get context from both chat history and vector database
+- `write_records()`: Write new records to both chat history and vector database
+- `get_context_creator()`: Get the context creator
+- `clear()`: Remove all records from both memory blocks
+
+## Usage
+
+To use the Memory module in your agent:
+
+1. Choose an appropriate AgentMemory implementation (ChatHistoryMemory, VectorDBMemory, or LongtermAgentMemory).
+2. Initialize the memory with a context creator and any necessary parameters.
+3. Use `write_records()` to add new information to the memory.
+4. Use `retrieve()` to get relevant context for the agent's next action.
+5. Use `get_context()` to obtain the formatted context for the agent.
+
+Example using LongtermAgentMemory:
+
+```python
+from camel.memories import LongtermAgentMemory, ChatHistoryBlock, VectorDBBlock, ScoreBasedContextCreator
+from camel.memories import MemoryRecord
+from camel.messages import BaseMessage
+from camel.types import OpenAIBackendRole
+from camel.utils import OpenAITokenCounter
+
+# Initialize the memory
+memory = LongtermAgentMemory(
+    context_creator=ScoreBasedContextCreator(OpenAITokenCounter, 1024),
+    chat_history_block=ChatHistoryBlock(),
+    vector_db_block=VectorDBBlock()
+)
+
+# Create and write new records
+records = [
+    MemoryRecord(
+        message=BaseMessage(role_name="user", role_type="human", meta_dict=None, content="What is machine learning?"),
+        role_at_backend=OpenAIBackendRole.USER
+    ),
+    MemoryRecord(
+        message=BaseMessage(role_name="agent", role_type="ai", meta_dict=None, content="Machine learning is a subset of AI..."),
+        role_at_backend=OpenAIBackendRole.ASSISTANT
+    )
+]
+memory.write_records(records)
+
+# Get context for the agent
+context, token_count = memory.get_context()
+
+# Use the context in your agent's next action
+
+```
+
+## Advanced Topics
+
+### Customizing Context Creator
+
+You can create custom context creators by subclassing `BaseContextCreator`:
+
+```python
+from camel.memories import BaseContextCreator
+
+class MyCustomContextCreator(BaseContextCreator):
+    @property
+    def token_counter(self):
+        # Implement your token counting logic
+        return 
+
+    @property
+    def token_limit(self):
+        return 1000  # Or any other limit
+
+    def create_context(self, records):
+        # Implement your context creation logic
+        pass
+
+```
+
+### Fine-tuning Vector Database
+
+For VectorDBBlock and VectorDBMemory, you can fine-tune the vector database by adjusting the embedding model or storage parameters:
+
+```python
+from camel.embeddings import mistral_embedding, openai_embedding
+from camel.storages.vectordb_storages import qdrant, milvus
+from camel.memories import VectorDBBlock
+
+vector_db1 = VectorDBBlock(
+    embedding = mistral_embedding,
+    storage = qdrant
+)
+
+vector_db2 = VectorDBBlock(
+    embedding=openai_embedding,
+    storage=milvus
+)
+```
```

Comment:
I think this code block does not run. 
The import should be `OpenAIEmbedding`, `MistralEmbedding`, etc.